lib/semantics/mod-file.cpp

```diff
@@ -618,53 +615,50 @@ std::ostream &PutLower(std::ostream &os, const std::string &str) {
 }
 
 struct Temp {
-  Temp() = delete;
+  Temp(llvm::sys::fs::file_t fd, std::string path) : fd{fd}, path{path} {}
+  Temp(Temp&& t) : fd{std::exchange(t.fd, -1)}, path{std::move(t.path)} {}
   ~Temp() {
-    close(fd);
-    unlink(path.c_str());
+    if (fd >= 0) {
+      llvm::sys::fs::closeFile(fd);
+      llvm::sys::fs::remove(path.c_str());
+    }
   }
-  int fd;
+  llvm::sys::fs::file_t fd;
   std::string path;
 };
 
 // Create a temp file in the same directory and with the same suffix as path.
 // Return an open file descriptor and its path.
-static Temp MkTemp(const std::string &path) {
+static llvm::ErrorOr<Temp> MkTemp(const std::string &path) {
   auto length{path.length()};
   auto dot{path.find_last_of("./")};
-  std::string suffix{dot < length && path[dot] == '.' ? path.substr(dot) : ""};
+  std::string suffix{
+      dot < length && path[dot] == '.' ? path.substr(dot + 1) : ""};
   CHECK(length > suffix.length() &&
       path.substr(length - suffix.length()) == suffix);
-  auto tempPath{path.substr(0, length - suffix.length()) + "XXXXXX" + suffix};
-  int fd{mkstemps(&tempPath[0], suffix.length())};
-  auto mask{umask(0777)};
-  umask(mask);
-  chmod(tempPath.c_str(), 0666 & ~mask);  // temp is created with mode 0600
-  return Temp{fd, tempPath};
+  auto prefix = path.substr(0, length - suffix.length());
+  llvm::sys::fs::file_t fd;
+  llvm::SmallString<16> tempPath;
+  if (std::error_code err = llvm::sys::fs::createUniqueFile(prefix + "%%%%%%" + suffix, fd, tempPath))
+    return err;
```

Comment:
Please put braces around the body of the if statement.

Reply:
And around the initialization in the `if` statement.